src/features/movies/moviesSaga.js

```diff
@@ -1,14 +1,14 @@
-import axios from "axios";
-import { takeLatest, put } from "redux-saga/effects";
-import { fetchError, fetchMoviesSuccess, fetchMovieSuccess } from "./moviesSlice";
-import { apiUrl, apiKey } from "../../apiData";
+// import axios from "axios";
+import { takeLatest, put, call, delay } from "redux-saga/effects";
+import { fetchError, fetchMoviesSuccess, fetchMovies } from "./moviesSlice";
+// import { apiUrl, apiKey } from "../../apiData";
+import { getMovies } from "./api";
 
-function* fetchMoviesHandler() {
+function* fetchMoviesHandler({ payload }) {
     try {
-        const data = yield axios.get(
-            `${apiUrl}movie/popular?api_key=${apiKey}&language=en-US&page=1`
-        );
+        const data = yield call(getMovies, payload);
```

Comment:
I think you can delete payload here. If I see correctly it's not used (at least not now).

Reply:
comments deleted